frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/moderation/ModerationPortletUtils.java

```diff
@@ -128,21 +129,28 @@ public static AddDocumentRequestSummary updateClearingRequest(PortletRequest req
                         && SW360Utils.isValidDate(preferredClearingDate, DateTimeFormatter.ISO_LOCAL_DATE, null)) {
                     clearingRequest.setRequestedClearingDate(preferredClearingDate);
                 }
+
+                String state = request.getParameter(ClearingRequest._Fields.CLEARING_STATE.toString());
+                ClearingRequestState requestState = ClearingRequestState.findByValue(parseInt(state));
                 if (CommonUtils.isNotNullEmptyOrWhitespace(isClearingExpertEdit) && Boolean.parseBoolean(isClearingExpertEdit)) {
                     String agreedDate = request.getParameter(ClearingRequest._Fields.AGREED_CLEARING_DATE.toString());
-                    String status = request.getParameter(ClearingRequest._Fields.CLEARING_STATE.toString());
                     String priority = request.getParameter(ClearingRequest._Fields.PRIORITY.toString());
                     String clearingType = request.getParameter(ClearingRequest._Fields.CLEARING_TYPE.toString());
                     if (CommonUtils.isNotNullEmptyOrWhitespace(agreedDate) && !agreedDate.equals(clearingRequest.getAgreedClearingDate())
                             && !SW360Utils.isValidDate(agreedDate, DateTimeFormatter.ISO_LOCAL_DATE, null)) {
                         log.warn("Invalid agreed clearing date: " + agreedDate + " is entered, by user: "+ user.getEmail());
                         return requestSummary.setMessage("Invalid agreed clearing date");
                     }
+                    clearingRequest.setClearingState(requestState);
                     clearingRequest.setAgreedClearingDate(CommonUtils.nullToEmptyString(agreedDate));
-                    clearingRequest.setClearingState(ClearingRequestState.findByValue(parseInt(status)));
                     clearingRequest.setPriority(ClearingRequestPriority.findByValue(parseInt(priority)));
                     clearingRequest.setClearingType(ClearingRequestType.findByValue(parseInt(clearingType)));
                 }
+                if (requestingUser.equals(user.getEmail())) {
```

Comment:
I think in this case, if the requestingUser has role CLEARING_EXPERT and the clearing state is CLOSED, the clearing request state will be set twice time.
Maybe you should replace **if** to **else if**